code/code/misc/combat.cc

```diff
@@ -2030,6 +2030,8 @@ static void checkLearnFromHit(TBeing * ch, int tarLevel, TThing * o, bool isPrim
 	else
 	  ch->learnFromDoingUnusual(LEARN_UNUSUAL_NORM_LEARN, SKILL_OFFENSE, (225 - (2* myLevel)));
       }
+      // Learn advanced offense
+      ch->learnFromDoingUnusual(LEARN_UNUSUAL_NORM_LEARN, SKILL_ADVANCED_OFFENSE, (170 - (2* myLevel)));
```

Comment:
Will it work if you don't have the skill?

Reply:
learnFromDoingUnusual has a check if the skill is known so you won't learn a skill you don't know.